shared/src/main/java/uk/gov/di/authentication/shared/entity/ClientSession.java

```diff
@@ -21,7 +22,9 @@ public class ClientSession {
 
     @Expose private VectorOfTrust effectiveVectorOfTrust;
 
-    @Expose private Subject docAppSubjectId;
+    @JsonAdapter(SubjectAdapter.class)
```

Comment:
I'd be tempted to globally register the adapter in the `SerializationService` (see L32 for how we did `StateAdapter`) ... this is behaviour we want anywhere GSON encounters a `Subject` rather than just this specific class.

Reply:
I've added it to the `SerializationService` but without this here, tests will fail as the `docAppSubjectId` will sometimes be null. Not sure how to get around this otherwise.